test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -1152,22 +1154,31 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 	})
 
-	// TODO - Need specific container runtime utility for Hypershift
-	Context("ContainerRuntimeConfig", Ordered, Label(string(label.Tier2), string(label.OpenShift)), func() {
+	Context("ContainerRuntimeConfig", Ordered, Label(string(label.Tier2)), func() {
 		var ctrcfg *machineconfigv1.ContainerRuntimeConfig
+		var ctrcfg_configmap *corev1.ConfigMap
 		const ContainerRuntimeConfigName = "ctrcfg-test"
 		mcp := &machineconfigv1.MachineConfigPool{}
 		var testpodTemplate *corev1.Pod
 		BeforeAll(func() {
 			key := types.NamespacedName{
 				Name: poolName,
 			}
-			Expect(testclient.Client.Get(context.TODO(), key, mcp)).ToNot(HaveOccurred(), "cannot get MCP %q", poolName)
-			By("checking if ContainerRuntimeConfig object already exists")
-			ctrcfg, err = getContainerRuntimeConfigFrom(context.TODO(), profile, mcp)
-			Expect(err).ToNot(HaveOccurred(), "failed to get ContainerRuntimeConfig from profile %q mcp %q", profile.Name, mcp.Name)
-			if ctrcfg != nil {
-				Skip(fmt.Sprintf("ContainerRuntimeConfig %s exist in the cluster and not expected", ctrcfg.Name))
+			if !hypershift.IsHypershiftCluster() {
+				Expect(testclient.ControlPlaneClient.Get(context.TODO(), key, mcp)).ToNot(HaveOccurred(), "cannot get MCP %q", poolName)
+				By("checking if ContainerRuntimeConfig object already exists")
+				ctrcfg, err = getContainerRuntimeConfigFrom(context.TODO(), profile, mcp)
+				Expect(err).ToNot(HaveOccurred(), "failed to get ContainerRuntimeConfig from profile %q mcp %q", profile.Name, mcp.Name)
+				if ctrcfg != nil {
+					Skip(fmt.Sprintf("ContainerRuntimeConfig %s exist in the cluster and not expected", ctrcfg.Name))
```

Comment:
@Tal-or Well this check was from the original test before even my refactor.
If we plan to use this e2e **solely** on CI and therefore no ctrcfg is expected, then I guess it is safe to remove this.  
However, if this test is somehow executed on a cluster with container runtime config it might run into trouble since it is not designed to start this way. I would not remove this check.

Reply:
Our tests are intended to run on clean, persisted and consistent environment. 
As I suggested to @SargunNarula we should not skip but log with error because having unexpected objects on the cluster, might hide even a bigger problem.

Reply:
Logging an error and running this test could potentially fail or not test the desired functionality. IOW, this test is incorrect if it runs on a system with ctrcfg pre-installed.
I think the best way to go forward here is to ensure we do not have "unexpected objects on the cluster, might hide even a bigger problem." is to fail the test right away if a ctrcfg is found with a proper message